File: app/src/main/java/me/dara/memoapp/ui/dashboard/MemoCreateFragment.java
package me.dara.memoapp.ui.dashboard;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;
import me.dara.memoapp.R;
import me.dara.memoapp.databinding.FragmentMemoCreateBinding;

/**
 * @author sardor
 */
public class MemoCreateFragment extends Fragment {

  DashBoardViewModel viewModel;
  FragmentMemoCreateBinding binding;

  public View onCreateView(LayoutInflater inflater, ViewGroup container,
      Bundle savedInstanceState) {
    binding = DataBindingUtil.inflate(inflater, R.layout.fragment_memo_create,
        container, true);
    return binding.getRoot();
  }

  @Override public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);
    viewModel = ViewModelProviders.of(this).get(DashBoardViewModel.class);

  }

  /*
  *
  *
  *
  *
  *
  *

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    viewModel = ViewModelProviders.of(this).get(DashBoardViewModel::class.java)
    imgMemo.setOnClickListener {
      startActivityForResult(ImageUtil.getPickImageIntent(requireContext()), 1000)
    }
    dateMemo.setOnClickListener {
      val picker = DatePickerAppCompat()
      picker.listener = { _: DatePicker?, year: Int, month: Int, dayOfMonth: Int ->
        Toast.makeText(requireContext(), "year:$year,month:$month,dayofMonth:$dayOfMonth", Toast.LENGTH_LONG).show()
      }
      picker.show(childFragmentManager, "DatePickerAppCompat")
    }
    btnCreateMem.setOnClickListener {
      viewModel.postUser(User("sardorislomov96gmailcom","5341","phot_url2",false,29))
    }
  }

  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
    if (requestCode == 1000) {
      val bmp = ImageUtil.getImageFromResult(requireContext(), resultCode, data)
      imgMemo.setImageBitmap(bmp)
    }
  }
  *
  *
  *
  *
  *
  *
  *
  * */
}

File: app/src/main/java/me/dara/memoapp/ui/dashboard/MemoCreateFragment.kt
package me.dara.memoapp.ui.dashboard

import android.app.DatePickerDialog
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.DatePicker
import android.widget.Toast
import kotlinx.android.synthetic.main.fragment_memo_create.*

import me.dara.memoapp.R
import me.dara.memoapp.ui.view.DatePickerAppCompat

class MemoCreateFragment : Fragment() {

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                            savedInstanceState: Bundle?): View? {
    // Inflate the layout for this fragment
    return inflater.inflate(R.layout.fragment_memo_create, container, false)
  }

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)

    imgMemo.setOnClickListener {

    }
    dateMemo.setOnClickListener {
      val picker = DatePickerAppCompat()
      picker.listener = { view: DatePicker?, year: Int, month: Int, dayOfMonth: Int ->
        Toast.makeText(requireContext(), "year:$year,month:$month,dayofMonth:$dayOfMonth", Toast.LENGTH_LONG).show()
      }
      picker.show(childFragmentManager, "DatePickerAppCompat")
    }


  }


  fun imgPicker() {

  }


}
